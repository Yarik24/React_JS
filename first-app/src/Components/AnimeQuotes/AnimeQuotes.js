import {CircularProgress, Dialog, DialogTitle, TextField, Button} from "@material-ui/core";
import { withStyles } from "@material-ui/styles"; 
import { memo, useEffect, useState } from "react";
import { shallowEqual, useSelector } from "react-redux";
import Header from "../Header/Header";
import InputForm from "../InputForm/InputForm";
import QuotesList from "../QuotesList/QuotesList";

function AnimeQuotes() {
  
  document.title = "Anime quotes";
  const [modalFlag, setModalFlag] = useState(false);
  const quotes = useSelector((state) => state.quotes.quotesList, shallowEqual);
  const loading = useSelector((state) => state.quotes.loading, shallowEqual);
  let error = useSelector((state) => state.quotes.error);

  useEffect(() => {
    if (error) {
      setModalFlag(true);
    }
  }, [error]);

  const StyledCircularProgress = withStyles({
    root: {
      position: "fixed",
      top: "calc(50% - 20px)",
      left: "calc(50% - 20px)",
    },
  })(CircularProgress);

  return (
    <>
      {loading && <StyledCircularProgress />}
      <Header />
      <QuotesList quotes={quotes} />
      <InputForm />
      <Dialog open={modalFlag}>
        <DialogTitle>Error</DialogTitle>
        <TextField value={error} sx={{ width: 250 }} />
        <Button onClick={() => setModalFlag(false)}>ะะบ</Button>
      </Dialog>
    </>
  );
}

export default memo(AnimeQuotes);