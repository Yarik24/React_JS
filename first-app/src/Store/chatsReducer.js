import { UPDATE_CHATS } from "./Types/chatsTypes";

const InitialState = { chatList: null };

function chatsReducer(state = InitialState, action) {
  switch (action.type) {
    // case ADD_CHAT:
    //   count++;
    // return {
    //   ...state,
    //   chatList: [
    //     ...state.chatList,
    //     {
    //       id: `id${count}`,
    //       name: action.payload,
    //     },
    //   ],
    // };

    // case DELETE_CHAT:
    //   return {
    //     ...state,
    //     chatList: state.chatList.filter((el) => el.id !== action.payload),
    //   };

    case UPDATE_CHATS:
      return { chatList: action.payload };

    default:
      return state;
  }
}

export default chatsReducer;