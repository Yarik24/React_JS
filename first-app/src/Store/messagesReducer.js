import { UPDATE_MESSAGE } from "./Types/messagesTypes";

const initialState = { messageList: null };

function messagesReducer(state = initialState, action) {
  switch (action.type) {
    case UPDATE_MESSAGE: {
      return { ...state, messageList: action.payload };
    }

    default:
      return state;
  }
}

export default messagesReducer;