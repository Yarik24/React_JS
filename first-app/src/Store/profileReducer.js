import CHANGE_NAME from "./Types/profileTypes";

const InitialState = { name: "No name" };

function ProfileReducer(state = InitialState, { type, payload }) {
  switch (type) {
    case CHANGE_NAME:
      return { name: payload };
    default:
      return state;
  }
}

export default ProfileReducer;