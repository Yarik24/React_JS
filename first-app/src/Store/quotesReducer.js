import { STATUSES } from "../Action/quotesAction";
import {GET_QUOTE_REQUEST, GET_QUOTE_SUCCESS, GET_QUOTE_FAILURE} from "./Types/quotesTypes";

const InitialState = {
  quotesList: [],
  loading: false,
  request: STATUSES.IDLE,
  error: null,
};

function quotesReducer(state = InitialState, action) {
  switch (action.type) {
    case GET_QUOTE_REQUEST:
      return {
        ...state,
        request: STATUSES.REQUEST,
        loading: true,
        error: null,
      };

    case GET_QUOTE_SUCCESS:
      return {
        ...state,
        quotesList: [...state.quotesList, action.payload],
        request: STATUSES.SUCCESS,
        loading: false,
      };

    case GET_QUOTE_FAILURE:
      return {
        ...state,
        request: STATUSES.FAILURE,
        error: action.payload,
        loading: false,
      };

    default:
      return state;
  }
}

export default quotesReducer;