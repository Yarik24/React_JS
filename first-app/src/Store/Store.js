import profileReducer from "./profileReducer";
import chatsReducer from "./chatsReducer";
import messagesReducer from "./messagesReducer";
import quotesReducer from "./quotesReducer";
import { createStore, combineReducers, applyMiddleware } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import thunk from "redux-thunk";
import { persistStore, persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage"; 

const persistConfig = { // Cоздаем объект конфигурации для persist
  key: "React App",
  storage,
  blacklist: ["quotes"]
};

const rootReducer = combineReducers({
  profile: profileReducer,
  chats: chatsReducer,
  messages: messagesReducer,
  quotes: quotesReducer
});

const persistedReducer = persistReducer(persistConfig, rootReducer); // оборачиваем редьюсеры в persist

export const store = createStore( // Cоздаем store с использованием persistedReducer
  persistedReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

export const persistor = persistStore(store); // Cоздаем persistor