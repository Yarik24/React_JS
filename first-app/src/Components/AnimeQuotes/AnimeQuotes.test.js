import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { store } from "../../Store";
import AnimeQuotes from "./AnimeQuotes";

it("Creating a snapshot of a component AnimeQuotes", () => {
  const component = render(
    <Provider store={store}>
      <BrowserRouter>
        <AnimeQuotes />
      </BrowserRouter>
    </Provider>
  );
  expect(component).toMatchSnapshot();
});