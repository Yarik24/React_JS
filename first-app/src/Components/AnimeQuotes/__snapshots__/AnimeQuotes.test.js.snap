exports[`matches snapshot with no article 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <ul
          class="header__links"
        >
          <li>
            <a
              href="/profile"
              style="text-decoration: none;"
            >
              Профиль
            </a>
          </li>
          <li>
            <a
              href="/chats"
              style="text-decoration: none;"
            >
              Чаты
            </a>
          </li>
          <li>
            <a
              href="/quotes"
              style="text-decoration: none;"
            >
              Аниме цитаты
            </a>
          </li>
        </ul>
        <div
          class="header__authBtns"
        >
          <button>
            Вход
          </button>
        </div>
      </header>
      <div
        class="quotes"
      >
        <p
          class="quotes__p"
        >
          Send a blank message to get a quote from a random anime. Or enter the name of the anime in Latin for a quote from a specific anime.
        </p>
        <ul
          class="quotes__ul"
        />
      </div>
      <form
        class="form"
      >
        <div
          class="MuiFormControl-root MuiTextField-root form__text css-p21bt9-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused css-bohyag-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="filled-basic-small"
            id="filled-basic-small-label"
          >
            Message
          </label>
          <div
            class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl MuiInputBase-sizeSmall css-cio0x1-MuiInputBase-root-MuiFilledInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
              id="filled-basic-small"
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root form__submit css-ke5b6m-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Send
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <ul
        class="header__links"
      >
        <li>
          <a
            href="/profile"
            style="text-decoration: none;"
          >
            Профиль
          </a>
        </li>
        <li>
          <a
            href="/chats"
            style="text-decoration: none;"
          >
            Чаты
          </a>
        </li>
        <li>
          <a
            href="/quotes"
            style="text-decoration: none;"
          >
           
           Anime quotes
          </a>
        </li>
      </ul>
      <div
        class="header__authBtns"
      >
        <button>
          Вход
        </button>
      </div>
    </header>
    <div
      class="quotes"
    >
      <p
        class="quotes__p"
      >
       
       Send a blank message to get a quote from a random anime. Or enter the name of the anime in Latin for a quote from a specific anime.
      </p>
      <ul
        class="quotes__ul"
      />
    </div>
    <form
      class="form"
    >
      <div
        class="MuiFormControl-root MuiTextField-root form__text css-p21bt9-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused css-bohyag-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="filled-basic-small"
          id="filled-basic-small-label"
        >
          Message
        </label>
        <div
          class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl MuiInputBase-sizeSmall css-cio0x1-MuiInputBase-root-MuiFilledInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
            id="filled-basic-small"
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root form__submit css-ke5b6m-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Send
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Creating a snapshot of a component AnimeQuotes 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <ul
          class="header__links"
        >
          <li>
            <a
              href="/profile"
              style="text-decoration: none;"
            >
              Profile
            </a>
          </li>
          <li>
            <a
              href="/chats"
              style="text-decoration: none;"
            >
              Chats
            </a>
          </li>
          <li>
            <a
              href="/quotes"
              style="text-decoration: none;"
            >
              Anime Quotes
            </a>
          </li>
        </ul>
        <div
          class="header__authBtns"
        >
          <button>
            Entrance
          </button>
        </div>
      </header>
      <div
        class="quotes"
      >
        <p
          class="quotes__p"
        >
          Send a blank message to get a quote from a random anime. Or enter the name of the anime in Latin for a quote from a specific anime.
        </p>
        <ul
          class="quotes__ul"
        />
      </div>
      <form
        class="form"
      >
        <div
          class="MuiFormControl-root MuiTextField-root form__text css-p21bt9-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused css-bohyag-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="filled-basic-small"
            id="filled-basic-small-label"
          >
            Message
          </label>
          <div
            class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl MuiInputBase-sizeSmall css-cio0x1-MuiInputBase-root-MuiFilledInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
              id="filled-basic-small"
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root form__submit css-ke5b6m-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Send
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <ul
        class="header__links"
      >
        <li>
          <a
            href="/profile"
            style="text-decoration: none;"
          >
            Profile
          </a>
        </li>
        <li>
          <a
            href="/chats"
            style="text-decoration: none;"
          >
            Chats
          </a>
        </li>
        <li>
          <a
            href="/quotes"
            style="text-decoration: none;"
          >
            Anime Quotes
          </a>
        </li>
      </ul>
      <div
        class="header__authBtns"
      >
        <button>
          Entrance
        </button>
      </div>
    </header>
    <div
      class="quotes"
    >
      <p
        class="quotes__p"
      >
        Send a blank message to get a quote from a random anime. Or enter the name of the anime in Latin for a quote from a specific anime.
      </p>
      <ul
        class="quotes__ul"
      />
    </div>
    <form
      class="form"
    >
      <div
        class="MuiFormControl-root MuiTextField-root form__text css-p21bt9-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused css-bohyag-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="filled-basic-small"
          id="filled-basic-small-label"
        >
          Message
        </label>
        <div
          class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl MuiInputBase-sizeSmall css-cio0x1-MuiInputBase-root-MuiFilledInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
            id="filled-basic-small"
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root form__submit css-ke5b6m-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Send
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;